src/github-api.ts

```diff
@@ -32,45 +19,92 @@ export interface Options {
     }
   }
 
+  async initOctokit() {
+    if (this.octokit) {
+      return;
+    }
+    const baseUrl = process.env.GITHUB_API_URL || `https://api.${this.github}`;
+    const throttling = await import("@octokit/plugin-throttling");
+    const Octokit = (await import("@octokit/rest")).Octokit.plugin(throttling.throttling);
+    this.octokit = new Octokit({
+      auth: this.auth,
+      baseUrl,
+      throttle: {
+        onRateLimit: (retryAfter, options, octokit, retryCount) => {
+          octokit.log.warn(`Request quota exhausted for request ${options.method} ${options.url}`);
+          octokit.log.info(`Retrying after ${retryAfter} seconds!`);
+          return true;
+        },
+        onSecondaryRateLimit: (retryAfter, options, octokit) => {
+          octokit.log.warn(`SecondaryRateLimit detected for request ${options.method} ${options.url}`);
+          octokit.log.info(`Retrying after ${retryAfter} seconds!`);
+          return true;
+        },
+      },
+    });
+  }
+
   public getBaseIssueUrl(repo: string): string {
     return `https://${this.github}/${repo}/issues/`;
   }
 
-  public async getIssueData(repo: string, issue: string): Promise<GitHubIssueResponse> {
-    const prefix = process.env.GITHUB_API_URL || `https://api.${this.github}`;
-    return this._fetch(`${prefix}/repos/${repo}/issues/${issue}`);
+  public async getPullRequest(repoFullName: string, commit: string) {
+    await this.initOctokit();
+    const [owner, repo] = repoFullName.split("/");
+    try {
+      const pullRequests = await this.octokit.repos.listPullRequestsAssociatedWithCommit({
+        owner,
+        repo,
+        commit_sha: commit,
+      });
+      if ((pullRequests.data as any)?.status) {
+        throw new Error(JSON.stringify((pullRequests.data as any).body));
+      }
+      return pullRequests.data?.[0];
+    } catch (e) {
+      if ((e as any).status === 404) {
+        return null;
+      }
+      if ((e as any).status === 422) {
+        return null;
+      }
+      throw e;
+    }
   }
 
-  public async getUserData(login: string): Promise<GitHubUserResponse> {
-    const prefix = process.env.GITHUB_API_URL || `https://api.${this.github}`;
-    return this._fetch(`${prefix}/users/${login}`);
-  }
+  public async getUserData(login: string) {
+    await this.initOctokit();
+    try {
+      const user = await this.octokit.users.getByUsername({
+        username: login,
+      });
 
-  private async _fetch(url: string): Promise<any> {
-    const res = await fetch(url, {
-      cachePath: this.cacheDir,
-      headers: {
-        Authorization: `token ${this.auth}`,
-      },
-    });
-    const parsedResponse = await res.json();
-    if (res.ok) {
-      return parsedResponse;
+      if ((user.data as any)?.status) {
+        throw new Error(JSON.stringify((user.data as any).body));
+      }
+      return user.data;
+    } catch (e) {
+      if ((e as any).status === 404) {
```

Comment:
so this is new functionality 🤔 it looks like the old behaviour was that it would throw an error here. Can you put it back to the old functionality here please?

Reply:
But then it would fail locally if any new commit is added and then try release plan

Reply:
I cannot mimic the old behaviour. The old behaviour was to error out if the PR does not exist.
But now, we do assume that we do not know the linked PR number to a commit and use the API to find it.
A commit without a PR can be valid